### PR TITLE
qownnotes: build with Qt6

### PR DESCRIPTION
**Pull request by totoroot**
###### Description of changes

- Build qownnotes with Qt6

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pbek** [comment]
You can drop qtxmlpatterns altogether, I removed the need for that dependency when porting the app to Qt6.

File: pkgs/top-level/all-packages.nix
Comment (wegank) [inline pkgs/top-level/all-packages.nix:11530]:
```suggestion
  qownnotes-qt6 = qt6Packages.callPackage ../applications/office/qownnotes {
    stdenv = if stdenv.isDarwin then darwin.apple_sdk_11_0.stdenv else stdenv;
  };
```

Comment (wegank) [reply pkgs/top-level/all-packages.nix:11530]: The `stdenv` override is needed for `libsForQt5.callPackage`, but not necessarily for `qt6Packages.callPackage`, I guess.

**Comment by totoroot** [comment]
Rebased onto master after version update got merged.
Removed dependency `qtxmlpatterns`.

File: pkgs/top-level/all-packages.nix
Comment (wineee) [inline pkgs/top-level/all-packages.nix:11529]: Renaming qownnotes to qownnotes-qt5 is unnecessary. Keeping the original name is fine, otherwise you would need to add aliases (`pkgs/top-level/aliases.nix`).

File: pkgs/applications/office/qownnotes/default.nix
Comment (wineee) [inline pkgs/applications/office/qownnotes/default.nix:64]: It seems that pbek is not in the [maintainer-list](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix).  @pbek  Would you like to add yourself to the maintainer list?

Comment (pbek) [reply pkgs/applications/office/qownnotes/default.nix:64]: Certainly, thank you for mentioning!
https://github.com/NixOS/nixpkgs/pull/227399

File: pkgs/top-level/all-packages.nix
Comment (pbek) [reply pkgs/top-level/all-packages.nix:11529]: @totoroot, @wineee, just for my understanding, is there an advantage in keeping both the qt5 and qt6 versions instead of just switching to qt6?

File: pkgs/top-level/all-packages.nix
Comment (wegank) [inline pkgs/top-level/all-packages.nix:11534]:
```suggestion
  qownnotes-qt6 = qt6Packages.callPackage ../applications/office/qownnotes { };
```

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000 Can you please review and let me know whether adding support for Qt6 like this is alright? Per default qownnotes would still be built using Qt5.

Why do we need both options? If qt6 just works, we can only offer that.

File: pkgs/top-level/all-packages.nix
Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:11529]:
```suggestion
  qownnotes = darwin.apple_sdk_11_0.callPackage ../applications/office/qownnotes {
```
for linux that wrapper is a noop

**Comment by pbek** [comment]
> > @SuperSandro2000 Can you please review and let me know whether adding support for Qt6 like this is alright? Per default qownnotes would still be built using Qt5.
> 
> Why do we need both options? If qt6 just works, we can only offer that.

Yeah, I was asking the same in https://github.com/NixOS/nixpkgs/pull/227301#discussion_r1173386878.
Would there be an advantage in keeping both, e.g. Better integration or disk space?

**Comment by totoroot** [comment]
I saw [this merged PR](https://github.com/NixOS/nixpkgs/pull/202103) and the [comment by @peterhoeg](https://github.com/NixOS/nixpkgs/pull/202103#issuecomment-1321389563) and thought that Qt6 might not be usable for some, so I wanted to be on the safe side like they did with `zeal`.

If there is no advantage in keeping both, then I will happily adapt the PR to only use Qt6.

**Comment by pbek** [comment]
And 32bit isn't supported by Qt6 any more...

> The issue with qt6 is that the applications look like crap under KDE because the theme isn't picked up automatically.
> https://github.com/NixOS/nixpkgs/pull/202103#issuecomment-1321389563

Yes, I feared as much.
In KDE Neon I had to UI issues with the Qt6 version of QOwnNotes.

**Review by pbek — APPROVED** [review]
qownnotes-qt6 builds and works nice. I coudn't find any issues.

**Comment by wegank** [comment]
Needs rebase at least.

**Comment by totoroot** [comment]
I'm still not sure what the correct approach would be here. I' don't know whether there has been a commitment by nixpkgs maintainers to move all possible packages to Qt6 packages once they build fine.

Like I said, I'd be happy to change the PR to only build QOwnNotes with Qt6 but would like to have some additional input.

**Comment by totoroot** [comment]
Since the consensus seems to be that there is no real problem in moving to building with Qt6, I changed the PR to only build with Qt6.
I guess we'll see whether someone complains about this move breaking something.

On NixOS with KDE Plasma QOwnNotes built with Qt6 looks just fine. I've also successfully built it on x86_64-darwin, and the only issue I've seen, is that the title bar is not adapting to my dark system theme, but I guess that's manageable.

**Comment by totoroot** [comment]
Not relevant to this PR, but I'm getting a bunch of warnings like this for these two plugins. @pbek are you aware of this?
```
Warning: In /nix/store/milahyvgak2ym1d5hk2zyl7pbljljjs7-qtwayland-5.15.8-bin/lib/qt-5.15.8/plugins/wayland-shell-integration/libwl-shell.so:
  Plugin uses incompatible Qt library (5.15.0) [release] ((null):0, (null))   nqqjc2vn-systemd-253.3-dev
...
Warning: In /nix/store/milahyvgak2ym1d5hk2zyl7pbljljjs7-qtwayland-5.15.8-bin/lib/qt-5.15.8/plugins/wayland-graphics-integration-client/libxcomposite-glx.so:
  Plugin uses incompatible Qt library (5.15.0) [release] ((null):0, (null))   1.29-dev/include -I/nix/st
```

**Comment by totoroot** [comment]
Result of `nixpkgs-review pr 227301` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>qownnotes</li>
  </ul>
</details>

**Comment by totoroot** [comment]
Result of `nixpkgs-review pr 227301` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>qownnotes</li>
  </ul>
</details>

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 227301` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>qownnotes</li>
  </ul>
</details>

**Comment by wegank** [comment]
> Not relevant to this PR, but I'm getting a bunch of warnings like this for these two plugins.
> 
> ```
> Warning: In /nix/store/milahyvgak2ym1d5hk2zyl7pbljljjs7-qtwayland-5.15.8-bin/lib/qt-5.15.8/plugins/wayland-shell-integration/libwl-shell.so:
>   Plugin uses incompatible Qt library (5.15.0) [release] ((null):0, (null))   nqqjc2vn-systemd-253.3-dev
> ...
> Warning: In /nix/store/milahyvgak2ym1d5hk2zyl7pbljljjs7-qtwayland-5.15.8-bin/lib/qt-5.15.8/plugins/wayland-graphics-integration-client/libxcomposite-glx.so:
>   Plugin uses incompatible Qt library (5.15.0) [release] ((null):0, (null))   1.29-dev/include -I/nix/st
> ```
> 
> @pbek are you aware of this?

This is not specific to `qownnotes`, see #227795.

**Comment by pbek** [comment]
Thank you!

**Comment by pbek** [comment]
I got similar (but different) warnings:

```
[08:52:26] [warning] In /nix/store/dlqrinrvpw234k2q1hsb5x4y0k90jw46-qtdeclarative-5.15.8-bin/lib/qt-5.15.8/qml/QtQml/Models.2/libmodelsplugin.so: Plugin uses incompatible Qt library (5.15.0) [release]
[08:52:26] [warning] In /nix/store/dlqrinrvpw234k2q1hsb5x4y0k90jw46-qtdeclarative-5.15.8-bin/lib/qt-5.15.8/qml/QtQml/WorkerScript.2/libworkerscriptplugin.so: Plugin uses incompatible Qt library (5.15.0) [release]
```

**Pull request merged**
